Source/Tools/FEXServer/CoreDumpService.h

```diff
@@ -0,0 +1,202 @@
+#include "Common/FEXServerClient.h"
+#include "Unwind/FileMapping.h"
+
+#include <FEXCore/fextl/map.h>
+#include <FEXCore/fextl/list.h>
+#include <FEXCore/fextl/string.h>
+#include <FEXCore/fextl/unordered_map.h>
+#include <FEXCore/fextl/vector.h>
+
+#include <cstring>
+#include <poll.h>
+#include <sys/socket.h>
+#include <thread>
+
+namespace Unwind {
+class Unwinder;
+}
+
+namespace CoreDumpService {
+  class CoreDumpClass {
+    public:
+      struct Description {
+        uint32_t pid;
+        uint32_t tid;
+        uint32_t uid;
+        uint32_t gid;
+        uint32_t Signal;
+        uint64_t Timestamp;
+        uint8_t HostArch;
+        uint8_t GuestArch;
+      };
+
+
+      int Init() {
+        int SVs[2];
+        int Result = socketpair(AF_UNIX, SOCK_STREAM | SOCK_CLOEXEC, 0, SVs);
+        if (Result == -1) {
+          return -1;
+        }
+
+        PollFDs.emplace_back(pollfd {
+          .fd = SVs[0],
+          .events = POLLIN | POLLHUP | POLLERR | POLLNVAL | POLLREMOVE | POLLRDHUP,
+          .revents = 0,
+        });
+
+        ExecutionThread = std::thread(&CoreDumpClass::ExecutionFunc, this);
+
+        ServerSocket = SVs[0];
+
+        // Return the second socket FD
+        return SVs[1];
+      }
+      ~CoreDumpClass() {
+        ExecutionThread.join();
+        close(ServerSocket);
+      }
+
+      void ExecutionFunc();
+
+      bool IsDone() {
+        return Running == false;
+      }
+
+      fextl::string const &GetCommandLineString() const {
+        return CommandLineString;
+      }
+
+      Description const &GetDescription() const {
+        return Desc;
+      }
+
+      FileMapping::FileMapping* FindFileMapping(const fextl::string &Path) {
+        auto it = PathToFileMap.find(Path);
+        if (it == PathToFileMap.end()) {
+          return nullptr;
+        }
+
+        return it->second;
+      }
+
+      FileMapping::FileMapping* GetFileMapping(uint64_t Addr) {
+        for (auto& Mapping : FileMappings) {
+          if (Mapping.Begin <= Addr && Mapping.End > Addr) {
+            return &Mapping;
+          }
+        }
+        return nullptr;
+      }
+
+      FileMapping::MemMapping const* GetMapping(uint64_t Addr) {
+        for (const auto& Mapping : MemMappings) {
+          if (Mapping.Begin <= Addr && Mapping.End > Addr) {
+            return &Mapping;
+          }
+        }
+        return nullptr;
+      }
+
+      bool AddressHasAccess(uint64_t Addr, bool Write) {
+        for (const auto& Mapping : MemMappings) {
+          if (Mapping.Begin <= Addr && Mapping.End > Addr) {
+            if (Write) {
+              return Mapping.permissions & (1U << 2);
+            }
+            else {
+              return Mapping.permissions & (1U << 3);
+            }
+          }
+        }
+
+        return false;
+      }
+
+      fextl::list<FileMapping::FileMapping> &GetFileMappingList() {
+        return FileMappings;
+      }
+
+    private:
+      void ParseCommandLineFD(int FD) {
```

Comment:
The current name got me scratching my head for a while, wondering why there's a dedicated FD for command line arguments. Can we rename this to `ParseCommandLineFromFD` to clarify this is just deserializing data expected in a specific format?

Ideally we'd rename the other functions in this group too, but at least let's fix it for functions added going forward.

Reply:
Renamed to `ParseCommandLineFromFD` and added a comment to it.